src/arithmetic/repository.c

```diff
@@ -6,57 +6,64 @@
 
 #include "repository.h"
 #include "../util/rmalloc.h"
-#include "../util/vector.h"
+#include "rax.h"
+#include <ctype.h>
+#include <assert.h>
 
 typedef struct {
-	const char *name;
 	AggFuncInit func;
 } __aggFuncEntry;
 
-static Vector *__aggRegisteredFuncs = NULL;
+static rax *__aggRegisteredFuncs = NULL;
+
+static void inline _toLower(const char *str, char *lower, short *lower_len) {
+	size_t str_len = strlen(str);
+	/* Avoid overflow. */
+	assert(*lower_len > str_len);
+
+	/* Update lower len*/
+	*lower_len = str_len;
+
+	int i = 0;
+	for(; i < str_len; i++) lower[i] = tolower(str[i]);
+	lower[i] = 0;
+}
 
 static void __agg_initRegistry() {
 	if(__aggRegisteredFuncs == NULL) {
-		__aggRegisteredFuncs = NewVector(__aggFuncEntry *, 8);
+		__aggRegisteredFuncs = raxNew();
 	}
 }
 
-int Agg_RegisterFunc(const char *name, AggFuncInit f) {
+void Agg_RegisterFunc(const char *name, AggFuncInit f) {
 	__agg_initRegistry();
 	__aggFuncEntry *e = rm_malloc(sizeof(__aggFuncEntry));
-	e->name = name;
 	e->func = f;
-	return Vector_Push(__aggRegisteredFuncs, e);
+	char lower_func_name[32] = {0};
+	short lower_func_name_len = 32;
+	_toLower(name, &lower_func_name[0], &lower_func_name_len);
+	raxInsert(__aggRegisteredFuncs, (unsigned char *)lower_func_name, strlen(lower_func_name), e, NULL);
```

Comment:
Replace `strlen(lower_func_name)` with `lower_func_name_len` as it is set by the `_toLower`

Reply:
correct!
fixed